src/diffpy/srmise/dataclusters.py

```diff
@@ -106,21 +130,26 @@ def setdata(self, x, y, res):
         # 3) r isn't sorted?
         if len(x) != len(y):
             raise ValueError("Sequences x and y must have the same length.")
-        if res <= 0:
-            raise ValueError("Resolution res must be greater than 0.")
+        if res < 0:
+            raise ValueError("Resolution res must be non-negative.")
         # Test for sorting?
-
         self.x = x
         self.y = y
         self.res = res
-
-        self.data_order = self.y.argsort()  # Defines order of clustering
-        self.clusters = np.array([[self.data_order[-1], self.data_order[-1]]])
-        self.current_idx = len(self.data_order) - 1
-        self.lastcluster_idx = 0
-        self.lastpoint_idx = self.data_order[-1]
-
-        self.status = self.READY
+        # If x sequence size is empty, set the object into Initialized state.
+        if x.size == 0 and res == 0:
```

Comment:
what  will happen if `x.size == 0` but `res != 0`?

Reply:
This is not what I meant.  I think that we want the clear method test to pass (which it was) but if we were to write tests for this method, there would be other situations than `x.size == 0 and res == 0` and these were not being handled in your code fix.

Let's just revert this commit and I will merge, since it is passing tests, but write tests for this setdata method on a separate PR.